app/Models/Timesheet.php

```diff
@@ -431,44 +431,90 @@ public static function retrieveUnconfirmedPeopleForYear($year)
-        if (empty($hoursEarned)) {
-            return [];
-        }
-
-        $hoursEarned = collect($hoursEarned);
-        $personIds = $hoursEarned->pluck('person_id');
-        $hoursByPerson = $hoursEarned->keyBy('person_id');
-
-        $people = Person::select('id', 'callsign', 'status', 'first_name', 'last_name', 'longsleeveshirt_size_style', 'teeshirt_size_style')
-                ->whereIn('id', $personIds)
-                ->where('status', Person::ACTIVE)
-                ->orderBy('callsign')
-                ->get();
-
-        return $people->map(function ($person) use ($thresholdSS, $thresholdLS, $hoursByPerson) {
-            $hours  = $hoursByPerson[$person->id]->seconds / 3600.00;
-
-            return [
-                'id'    => $person->id,
-                'callsign'  => $person->callsign,
-                'first_name' => $person->first_name,
-                'last_name' => $person->last_name,
-                'status'    => $person->status,
-                'email'     => $person->email,
-                'longsleeveshirt_size_style' => $person->longsleeveshirt_size_style,
-                'earned_ls' => ($hours >= $thresholdLS),
-                'teeshirt_size_style' => $person->teeshirt_size_style,
-                'earned_ss' => ($hours >= $thresholdSS), // gonna be true always, but just in case the selection above changes.
-                'hours'   => round($hours, 2),
-            ];
-        });
+      $active_statuses = implode("','", Person::ACTIVE_STATUSES);
+      $report = DB::select(
+       "SELECT
+          person.id, person.callsign, person.status, person.first_name, person.mi, person.last_name,
+          eh.estimated_hours,
+          ah.actual_hours,
+          person.teeshirt_size_style, person.longsleeveshirt_size_style
+        FROM
+          person
+        LEFT JOIN (
+          SELECT
+            person_slot.person_id,
+            round(sum(((TIMESTAMPDIFF(MINUTE, slot.begins, slot.ends))/60)),2) AS estimated_hours
+          FROM
+            slot
+          JOIN
+            person_slot ON person_slot.slot_id = slot.id
+          JOIN
+            position ON position.id = slot.position_id
+          WHERE
+            YEAR(slot.begins) = ?
+            AND position.count_hours IS TRUE
+          GROUP BY person_id
+        ) eh ON eh.person_id = person.id
+
+        LEFT JOIN (
+          SELECT
+            timesheet.person_id,
+            round(sum(((TIMESTAMPDIFF(MINUTE, timesheet.on_duty, timesheet.off_duty))/60)),2) AS actual_hours
+          FROM
+            timesheet
+          JOIN
+            position ON position.id = timesheet.position_id
+          WHERE
+            YEAR(timesheet.on_duty) = ?
+            AND position.count_hours IS TRUE
+          GROUP BY person_id
+        ) ah ON ah.person_id = person.id
+
+        WHERE
+          ( actual_hours >= ? OR estimated_hours >= ? )
```

Comment:
The query needs to check for ANY hours scheduled or actually worked - not if they reach a certain threshold. (i.e. "actual_hours > 0 or estimated_hours > 0")